engine/src/flutter/shell/platform/android/external_view_embedder/external_view_embedder_2.h

```diff
@@ -0,0 +1,157 @@
+#define FLUTTER_SHELL_PLATFORM_ANDROID_EXTERNAL_VIEW_EMBEDDER_EXTERNAL_VIEW_EMBEDDER_2_H_
+
+#include <unordered_map>
+
+#include "flutter/common/task_runners.h"
+#include "flutter/flow/embedded_views.h"
+#include "flutter/shell/platform/android/context/android_context.h"
+#include "flutter/shell/platform/android/external_view_embedder/surface_pool.h"
+#include "flutter/shell/platform/android/jni/platform_view_android_jni.h"
+#include "flutter/shell/platform/android/surface/android_surface.h"
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// Allows to embed Android views into a Flutter application.
+///
+/// This class calls Java methods via |PlatformViewAndroidJNI| to manage the
+/// lifecycle of the Android view corresponding to |flutter::PlatformViewLayer|.
+///
+/// It also orchestrates overlay surfaces. These are Android views
+/// that render above (by Z order) the Android view corresponding to
+/// |flutter::PlatformViewLayer|.
+///
+/// This implementation of the external view embedder is designed only to use
+/// HC++ mode. Mixing old HC modes is not supported, but either of the texture
+/// composition based platform views can be used with either mode.
+class AndroidExternalViewEmbedder2 final : public ExternalViewEmbedder {
+ public:
+  AndroidExternalViewEmbedder2(
+      const AndroidContext& android_context,
+      std::shared_ptr<PlatformViewAndroidJNI> jni_facade,
+      std::shared_ptr<AndroidSurfaceFactory> surface_factory,
+      const TaskRunners& task_runners);
+
+  // |ExternalViewEmbedder|
+  void PrerollCompositeEmbeddedView(
+      int64_t view_id,
+      std::unique_ptr<flutter::EmbeddedViewParams> params) override;
+
+  // |ExternalViewEmbedder|
+  DlCanvas* CompositeEmbeddedView(int64_t view_id) override;
+
+  // |ExternalViewEmbedder|
+  void SubmitFlutterView(
+      int64_t flutter_view_id,
+      GrDirectContext* context,
+      const std::shared_ptr<impeller::AiksContext>& aiks_context,
+      std::unique_ptr<SurfaceFrame> frame) override;
+
+  // |ExternalViewEmbedder|
+  PostPrerollResult PostPrerollAction(
+      const fml::RefPtr<fml::RasterThreadMerger>& raster_thread_merger)
+      override;
+
+  // |ExternalViewEmbedder|
+  DlCanvas* GetRootCanvas() override;
+
+  // |ExternalViewEmbedder|
+  void BeginFrame(GrDirectContext* context,
+                  const fml::RefPtr<fml::RasterThreadMerger>&
+                      raster_thread_merger) override;
+
+  // |ExternalViewEmbedder|
+  void PrepareFlutterView(SkISize frame_size,
+                          double device_pixel_ratio) override;
+
+  // |ExternalViewEmbedder|
+  void CancelFrame() override;
+
+  // |ExternalViewEmbedder|
+  void EndFrame(bool should_resubmit_frame,
+                const fml::RefPtr<fml::RasterThreadMerger>&
+                    raster_thread_merger) override;
+
+  bool SupportsDynamicThreadMerging() override;
```

Comment:
The above overrides all indicate the base class |ExternalViewEmbedder| should these last two overrides have the same comment?

Reply:
Fixed